Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+                var url = HMD.active
+                    ? OVERLAY_URL_SITTABLE_HMD
+                    : OVERLAY_URL_SITTABLE_DESKTOP;
+
+                overlaySittable = Overlays.addOverlay("image3d", {
+                    position: { x: 0.0, y: 0.0, z: 0.0 },
+                    rotation: Quat.multiply(chairProperties.rotation, Quat.fromVec3Degrees({ x: -90, y: 180, z: 0 })),
+                    dimensions: {
+                        x: 0.1,
+                        y: 0.1
+                    },
+                    url: url,
+                    ignoreRayIntersection: false,
+                    alpha: OVERLAY_SITTABLE_ALPHA_START,
+                    visible: true,
+                    emissive: true
+                });
+
+                var overlayDimensions = {
+                    x: 0.3,
+                    y: 0.3
+                };
+
+                var overlayPosition = Vec3.sum(chairProperties.position, { x: 0, y: 0, z: 0 });
+
+                Overlays.editOverlay(overlaySittable, {
+                    position: overlayPosition,
+                    dimensions: overlayDimensions
+                });
+
+            },
+            remove: function () {
+                if (overlaySittable !== null) {
+                    Overlays.deleteOverlay(overlaySittable);
+                    overlaySittable = null;
+                }
+            },
+            shouldShow: function () {
+                var seatPosition = Entities.getEntityProperties(entityID, ["position"]).position;
+                var distanceFromSeat = Vec3.distance(MyAvatar.position, seatPosition);
+                return distanceFromSeat < OVERLAY_SITTABLE_DISTANCE_SHOW && !utils.checkSeatForAvatar();
+            },
+            // This part is used to fade out the overlay over time
+            lerpTransparency: function () {
+                var startAlpha = OVERLAY_SITTABLE_ALPHA_START;
+                var changeAlpha = 0.01;
+                overlayIntervalTransparency = Script.setInterval(function () {
+                    startAlpha = startAlpha - changeAlpha; // My new alpha
+                    Overlays.editOverlay(overlaySittable, { alpha: startAlpha }); // Edit the existing overlay
+                    if (startAlpha <= OVERLAY_SITTABLE_MIN_ALPHA) {
+                        Script.clearInterval(overlayIntervalTransparency); // Stop my interval when it's faded out
+                        overlayIntervalTransparency = null;
+                    }
+                }, OVERLAY_SITTABLE_FADE);
+            },
+        },
+
+        preSit: {
+            preload: function () {
+                var str;
+                for (var i = 0; i < OVERLAY_PRESIT_URL_NUM; i++) {
+                    str = i + 1;
+                    overlayPreSitLoaded[i] = TextureCache.prefetch(OVERLAY_PRESIT_URL_ROOT + str + OVERLAY_PRESIT_URL_POSTFIX);
+                }
+                OVERLAY_PRESIT_URL_TEXT = TextureCache.prefetch(OVERLAY_PRESIT_URL_TEXT);
```

Comment:
This is bad form, in our coding convention standard a variable with ALL_CAPS is assumed to be a constant.  but here are you are changing it.  I would use a new varaible named overlayPresetTexture or something instead instead of changing the value of OVERLAY_PRESIT_URL_TEXT.